wolfcrypt/src/asn.c

```diff
@@ -11128,23 +11137,29 @@ int wc_EccPublicKeyDecode(const byte* input, word32* inOutIdx,
                                             key->heap, DYNAMIC_TYPE_ECC_BUFFER);
         }
         if (ret == 0) {
-            if (input[*inOutIdx] == ASN_BIT_STRING) {
+            if (*inOutIdx < inSz && input[*inOutIdx] == ASN_BIT_STRING) {
                 len = 0;
                 ret = GetASNHeader(input, ASN_BIT_STRING, inOutIdx, &len, inSz);
-                inOutIdx += len;
+                *inOutIdx += len;
             }
         }
         if (ret == 0) {
             ret = ASNToHexString(input, inOutIdx, (char**)&point, inSz,
                                             key->heap, DYNAMIC_TYPE_ECC_BUFFER);
+            if (ret == 0 && (int)XSTRLEN(point) < (curve->size * 4) + 2) {
```

Comment:
What does 4 mean, why?  What does 2 mean, why?  Is 2 just for the null termination later and may not be here?

Reply:
The values where added for checking before copying to curve->Gx and curve->Gy later on in the function. Each XMEMCPY is of size (curve->size * 2) and the first copy to curve->Gx is at an offset of 2 while the second copy (being to curve->Gy) is at an offset of (curve->size * 2 + 2).

I think the offset of 2 is to account for the first 2 hex values 0 & 4 prepended to the public ECC key. Will add in a comment to the code for the check.